src/tasks/generateSignatures.ts

```diff
@@ -0,0 +1,102 @@
+import fs from "fs";
+
+import "hardhat-deploy";
+import "@nomiclabs/hardhat-ethers";
+import { task } from "hardhat/config";
+
+import { TypedDataDomain } from "../../src/ts/ethers";
+
+export function domain(
+  chainId: number,
+  verifyingContract: string,
+): TypedDataDomain {
+  return {
+    name: "AccessManager",
+    version: "v1",
+    chainId,
+    verifyingContract,
+  };
+}
+import { getAllowListOffChainManagedContract } from "./utils";
+
+// interface Signature {
+//   address: string;
+//   signature: string;
+// }
+
+// interface SignatureSet {
+//   auctionId: number;
+//   chainId: number;
+//   signatures: Signature;
+// }
+
+const generateSignatures: () => void = () => {
+  task(
+    "generateSignatures",
+    "Generates the signatures for the allowListManager",
+  )
+    .addParam("auctionId", "Id of the auction ")
+    .addParam(
+      "fileWithAddress",
+      "File with comma separated addresses that should be allow-listed",
+    )
+    .setAction(async (taskArgs, hardhatRuntime) => {
+      const [caller] = await hardhatRuntime.ethers.getSigners();
+      console.log(
+        "Using the account: ",
+        caller.address,
+        " to generate signatures",
+      );
+      const allowListContract = await getAllowListOffChainManagedContract(
+        hardhatRuntime,
+      );
+      const { chainId } = await hardhatRuntime.ethers.provider.getNetwork();
+
+      const contractDomain = domain(chainId, allowListContract.address);
+
+      const file = fs.readFileSync(taskArgs.fileWithAddress, "utf8");
+      const addresses = file.split(",");
+      console.log(addresses);
+      const signatures = [];
+      for (const address of addresses) {
+        console.log("signing ", address);
```

Comment:
A nice feature could be printing everything that is not part of the JSON to stderr (`console.warn`), this way you can redirect the result of calling this function to a file without intermediate manual editing and you will still see what's going on:
```
npx hardhat generateSignatures ...args > signatures.json
0x00..., 0x12...
signing 0x00...
signing 0x12...
```
You could also do more fancy logging, where the output appears only if some debug flag is set.

Reply:
Unfortunately, this is not really working, as we also have other standard output.
My running of the script looks like this:
```
yarn run v1.22.4
$ /Users/alexherrmann/easyAuction/node_modules/.bin/hardhat generateSignatures --auction-id 1 --file-with-address ./test.txt --network rinkeby
Using the account:  0xfB696e9E9e5038DDc78592082689B149AB3a19d5  to generate signatures
{"auctionId":"1","chainId":4,"allowListContract":"0xed52BE1b0071C2f27D10fCc06Ef2e0194cF4E18D","signatures":["{\"address\":\"0x740a98F8f4fAe0986FB3264Fe4aaCf94ac1EE96f\",\"signature\":\"0x000000000000000000000000000000000000000000000000000000000000001b6e5cf2c8aad4817e6fdd674bdfb82ab3aeff34c6a5b26d238f79d8173e7d62714704d23bd2632ff6e1682567ed7fbe943694ec45810d5f87cd93828063fead8a\"}","{\"address\":\"0x04668ec2f57cc15c381b461b9fedab5d451c8f7f\",\"signature\":\"0x000000000000000000000000000000000000000000000000000000000000001c0b248f8378c255c3c355b3e4608636f84e26df8e1a8e1975c5ddad85f2f2dacb746ea0232445018cb2b8d174ce333d4b3b06c0e21548a4a77f727dc4051c2e25\"}"]}
Done in 3.93s.

```

Reply:
If the extra output is `yarn run ...`, `$ ...`, `Done in...`, then running yarn with `--silent` should suppress it:
```
yarn --silent hardhat generateSignatures ...
```